opm/grid/cpgrid/CpGridData.cpp

```diff
@@ -2037,6 +2069,76 @@ bool CpGridData::patchesShareFace(const std::vector<std::array<int,3>>& startIJK
     return false;
 }
 
+int CpGridData::sharedFaceTag(const std::vector<std::array<int,3>>& startIJK_2Patches, const std::vector<std::array<int,3>>& endIJK_2Patches) const
+{
+    assert(startIJK_2Patches.size() == 2);
+    assert(endIJK_2Patches.size() == 2);
+
+    int faceTag = -1; // 0 represents I_FACE, 1 J_FACE, and 2 K_FACE. Use -1 for no sharing face case.
+     
+    if (patchesShareFace(startIJK_2Patches, endIJK_2Patches)) {
+        
+        const auto& detectSharing = [](std::vector<int> faceIdxs, std::vector<int> otherFaceIdxs){
+            bool faceIsShared = false;
+            for (const auto& face : faceIdxs) {
+                for (const auto& otherFace : otherFaceIdxs) {
+                    faceIsShared = faceIsShared || (face == otherFace);
+                    if (faceIsShared) {
+                        return faceIsShared; // should be true here
+                    }
+                }
+            }
+            return faceIsShared; // should be false here
+        };
+     
+        const auto& [iFalse, iTrue, jFalse, jTrue, kFalse, kTrue] = this->getBoundaryPatchFaces(startIJK_2Patches[0], endIJK_2Patches[0]);
+        const auto& [iFalseOther, iTrueOther, jFalseOther, jTrueOther, kFalseOther, kTrueOther] =
+            this->getBoundaryPatchFaces(startIJK_2Patches[1], endIJK_2Patches[1]);
+
+
+        bool isShared = false;
+           
+        if (startIJK_2Patches[1][0] == endIJK_2Patches[0][0]) {
+            isShared = isShared || detectSharing(iTrue, iFalseOther);
+            if (isShared) {
+                faceTag = 0;
+            }   
+        }
+        if (endIJK_2Patches[1][0] == startIJK_2Patches[0][0]) {
+            isShared = isShared || detectSharing(iFalse, iTrueOther);
+            if (isShared) {
+                faceTag = 0;
+            }  
+        }
+        if (startIJK_2Patches[1][1] == endIJK_2Patches[0][1]) {
+            isShared = isShared || detectSharing(jTrue, jFalseOther);
+            if (isShared) {
+                faceTag = 1;
+            }  
+        }
+        if (endIJK_2Patches[1][1] == startIJK_2Patches[1][1]) {
+            isShared = isShared || detectSharing(jFalse, jTrueOther);
+            if (isShared) {
+                faceTag = 1;
+            }  
+        }
+        if (startIJK_2Patches[1][2] == endIJK_2Patches[0][2]) {
+            isShared = isShared || detectSharing(kTrue, kFalseOther);
+            if (isShared) {
+                faceTag = 2;
+            }  
+        }
+        if (endIJK_2Patches[1][2] == startIJK_2Patches[0][2]) {
+            isShared = isShared || detectSharing(kFalse, kTrueOther);
+            if (isShared) {
+                faceTag = 2;
+            }  
+        }
```

Comment:
Is this really needed? Shouldn't detectSharing be symmetric in the sense that `detectSharing(a, b) ==detectSharing(b, a)` holds?

This might also holds for every second if statement above

Reply:
As you mentioned, detectSharing(a,b) == detectSharing(b,a). Here we have 12 (potentially) different sets of face indices 
- 6 for patch_1 (all the faces laying on the left, right, front, back, bottom, and top boundary of the patch_1) 
and 
- other 6 for patch_2 (all the faces laying on the left, right, front, back, bottom, and top boundary of patch_2).
We have 6 if-statements since we check if patch1 is on the left, right, front, back, bottom, top of patch2  (and, for symmetry, we do not check the same 6 cases for patch2).  

Am I still repeating cases and I do not see it?  In any case, I'll add this description as a comment in the body function. Let me know if I still messed it up with extra cases/checks!

(I found a typo there too, so thanks for pointing at this piece of code!)